src/main/java/uk/gov/hmcts/reform/slc/services/ReportParser.java

```diff
@@ -31,11 +30,14 @@ public ParsedReport parse(Report report) {
 
             List<LetterPrintStatus> statuses =
                 stream(parser.spliterator(), false)
-                    .map(row -> toPrintStatus(row))
-                    .filter(Objects::nonNull)
+                    .map(csvRecord -> toPrintStatus(csvRecord))
                     .collect(toList());
```

Comment:
avoid collecting. unless declared stream will change in few lines below when filtering etc.

Reply:
need to collect, this collection is filtered later